src/middleware/with-logger.ts

```diff
@@ -0,0 +1,26 @@
+import { Middleware } from "./types.ts"
+
+export type Logger = {
+  debug: (...args: any[]) => void
+  info: (...args: any[]) => void
+  warn: (...args: any[]) => void
+  error: (...args: any[]) => void
+}
+
+/**
+ * Attaches a provided logger to ctx.logger.
+ * `ctx.logger` is used by internal EdgeSpec middleware when provided (instead of `console`).
+ */
+export const createWithLogger =
+  (
+    logger: Logger
```

Comment:
Since their logger object might have other options (like ours in `seam-connect`), can this be a generic instead? I.e. `<L extends Logger>`. 

Or maybe for best naming the interface becomes `LoggerInterface` or `LoggingFacade`. So then it's `<Logger extends LoggerInterface>`

Reply:
my vision was that people would always manually map the interface to their logger methods as the naming may not be 1:1, but making it a generic can't hurt

Reply:
yeah, fair point